src/update-changelog.ts

```diff
@@ -213,43 +264,17 @@ export async function updateChangelog({
     packageRename,
   });
 
-  // Ensure we have all tags on remote
-  await runCommand('git', ['fetch', '--tags']);
   const mostRecentTag = await getMostRecentTag({
     tagPrefixes,
   });
 
-  const commitRange =
-    mostRecentTag === null ? 'HEAD' : `${mostRecentTag}..HEAD`;
-  const commitsHashesSinceLastRelease = await getCommitHashesInRange(
-    commitRange,
-    projectRootDirectory,
-  );
-  const commits = await getCommits(commitsHashesSinceLastRelease);
-
-  const loggedPrNumbers = getAllLoggedPrNumbers(changelog);
-  const newCommits = commits.filter(
-    ({ prNumber }) =>
-      prNumber === undefined || !loggedPrNumbers.includes(prNumber),
-  );
-
-  const hasUnreleasedChanges =
-    Object.keys(changelog.getUnreleasedChanges()).length !== 0;
-  if (
-    newCommits.length === 0 &&
-    (!isReleaseCandidate || hasUnreleasedChanges)
-  ) {
-    return undefined;
-  }
-
   if (isReleaseCandidate) {
     if (!currentVersion) {
       throw new Error(
         `A version must be specified if 'isReleaseCandidate' is set.`,
       );
     }
-
-    if (mostRecentTag === `${tagPrefixes[0]}${currentVersion ?? ''}`) {
+    if (mostRecentTag === `${tagPrefixes[0]}${currentVersion}`) {
```

Comment:
`currentVersion` is narrowed to `string` thanks to the `if (!currentVersion)` block being moved directly above. Previously `as string` or `?? ''` was required.